packages/react-components/src/components/chart/eChartsWrapper.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef } from 'react';
+import { init, getInstanceByDom } from 'echarts';
+
+import type { ECharts, EChartsOption, SetOptionOpts } from 'echarts';
+
+export interface EChartsWrapperProps {
+  option: EChartsOption;
+  settings?: SetOptionOpts;
+  loading?: boolean;
+  theme?: string;
+  className?: string;
+  size?: { width?: number; height?: number };
+}
+
+export const EChartsWrapper = ({ option, settings, loading, theme, size = {} }: EChartsWrapperProps): JSX.Element => {
+  const chartRef = useRef<HTMLDivElement>(null);
+
+  // Initialize chart + listeners
+  useEffect(() => {
```

Comment:
I think we should turn these into hooks. I don't think we'd need the getInstanceByDom if we turn this into something like useECharts and it returns the chart?

Reply:
could probably also stuff all the resizing sideeffects in there too